yt/data_objects/selection_objects/data_selection_objects.py

```diff
@@ -41,15 +42,14 @@
 
 
 class YTSelectionContainer(YTDataContainer, ParallelAnalysisInterface, abc.ABC):
-    _locked = False
-    _sort_by = None
-    _selector = None
-    _current_chunk = None
-    _data_source = None
+    _locked: bool = False
+    _selector: SelectorObject | None = None
+    _current_chunk: YTDataContainer | None = None
+    _data_source: "YTSelectionContainer | None" = None
```

Comment:
A note here to reviewers: I _think_ `_data_source` cannot be `YTDataContainer`, since it is the base class and is never subclassed except for here. In other words, `YTSelectionContainer` is the most general use-facing class.

We could also implement `get_bbox` as an abstract method to `YTDataContainer` to force it to be defined everywhere.